pyccl/halos/profiles.py

```diff
@@ -267,17 +275,167 @@ def cumul2d(self, cosmo, r_t, M, a, mass_def=None):
-        if getattr(self, '_cumul2d', None):
+        if getattr(self, "_cumul2d", None):
             s_r_t = self._cumul2d(cosmo, r_t, M, a, mass_def)
         else:
-            s_r_t = self._projected_fftlog_wrap(cosmo, r_t, M,
-                                                a, mass_def,
-                                                is_cumul2d=True)
+            s_r_t = self._projected_fftlog_wrap(
+                cosmo, r_t, M, a, mass_def, is_cumul2d=True
+            )
         return s_r_t
 
-    def _fftlog_wrap(self, cosmo, k, M, a, mass_def,
-                     fourier_out=False,
-                     large_padding=True):
+    def sigma_critical(self, cosmo, a_lens, a_src):
+        """Returns the critical surface mass density.
+
+        .. math::
+         \\Sigma_{\\mathrm{crit}} = \\frac{c^2}{4\\pi G}
+          \\frac{D_{\\rm{s}}}{D_{\\rm{l}}D_{\\rm{ls}}},
+           where :math:`c` is the speed of light, :math:`G` is the
+           gravitational constant, and :math:`D_i` is the angular diameter
+           distance. The labels :math:`i =` s, l and ls denotes the distances
+           to the source, lens, and between source and lens, respectively.
+            Args:
+            cosmo (:class:`~pyccl.core.Cosmology`): A Cosmology object.
+            a_lens (float): lens' scale factor.
+            a_src (float or array_like): source's scale factor.
+
+        Returns:
+            float or array_like: :math:`\\Sigma_{\\mathrm{crit}}` in units
+            of :math:`\\M_{\\odot}/Mpc^2`
+        """
+        Ds = angular_diameter_distance(cosmo, a_src, a2=None)
+        Dl = angular_diameter_distance(cosmo, a_lens, a2=None)
+        Dls = angular_diameter_distance(cosmo, a_lens, a_src)
+        A = (
+            physical_constants.CLIGHT ** 2
+            * physical_constants.MPC_TO_METER
+            / (4.0 * np.pi * physical_constants.GNEWT
+               * physical_constants.SOLAR_MASS)
+        )
+
+        Sigma_crit = A * Ds / (Dl * Dls)
+        return Sigma_crit
+
+    def convergence(self, cosmo, r, M, a_lens, a_src, mass_def=None):
+        """ Returns the convergence as a function of cosmology,
+        radius, halo mass and the scale factors of the source
+        and the lens.
+
+        .. math::
+           \\kappa(R) = \\frac{\\Sigma(R)}{\\Sigma_{\\mathrm{crit}}},\\
+
+        where :math:`\\Sigma(R)` is the 2D projected surface mass density.
+
+        Args:
+            cosmo (:class:`~pyccl.core.Cosmology`): A Cosmology object.
+            r (float or array_like): comoving radius in Mpc.
+            M (float or array_like): halo mass in umits of M_sun.
+            a_lens (float or array_like): lens' scale factor.
+            a_src (float or array_like): source's scale factor.
+            mass_def (:class:`~pyccl.halos.massdef.MassDef`):
+                a mass definition object.
+
+        Returns:
+            float or array_like: convergence \
+                :math:`\\kappa`
+        """
+
```

Comment:
Blank lines are only used after class docstrings, not after the function docstrings.